local-volume/bootstrapper/bootstrapper.go

```diff
@@ -67,13 +71,48 @@ func setupClient() *kubernetes.Clientset {
 	return clientset
 }
 
-func generateMountName(hostDir, mountDir string) string {
+func generateMountName(mount *common.MountConfig) string {
 	h := fnv.New32a()
-	h.Write([]byte(hostDir))
-	h.Write([]byte(mountDir))
+	h.Write([]byte(mount.HostDir))
+	h.Write([]byte(mount.MountDir))
 	return fmt.Sprintf("mount-%x", h.Sum32())
 }
 
+// generateMountDir generates mount directory path in container, rule is to trim '/' prefix and
+// change "/" to "~", according to kubernetes convention, then concatenate with default root path.
+// E.g.
+//   "/mnt/ssds" -> "/mnt/local-storage/mnt~ssds"
+func generateMountDir(mount *common.MountConfig) string {
+	return path.Join(*mountRoot, strings.Replace(strings.TrimPrefix(mount.HostDir, "/"), "/", "~", -1))
```

Comment:
If we handle this, which part should we truncate or should we create ids?  There is also a corner case where it conflicts with other path if we truncate..

This is not common so i think we can ignore it for now